### ISSUE TITLE
query_log.query_start_time represents the query start time on the initiator

### ISSUE THREAD

**Issue by SaltTan** [ClickHouse/ClickHouse#59608]
It seems that for a distributed query shards record into `query_log` the event time from the initiator as the start time of the query, not the time when they received the request and started their own query.
As a result the start time for the local query is lost.
I expect shards to record their own timestamps as the initial stage of a distributed query can be quite long (the initiator might need to prepare a dataset to send with the query, for example).

```
query_id                             | type        | event_time_microseconds    | query_start_time_microseconds | query_duration_ms | is_initial_query | initial_query_id                    
-------------------------------------+-------------+----------------------------+-------------------------------+-------------------+------------------+-------------------------------------
b1c37b4c-3ef3-4026-8888-8facba026268 | QueryStart  | 2024-02-05 14:54:01.083110 | 2024-02-05 14:54:01.083110    |                 0 |                1 | b1c37b4c-3ef3-4026-8888-8facba026268
bec13910-2293-4226-bb2f-64cbac3b38a1 | QueryStart  | 2024-02-05 14:54:01.083110 | 2024-02-05 14:54:01.083110    |                 0 |                0 | b1c37b4c-3ef3-4026-8888-8facba026268
bec13910-2293-4226-bb2f-64cbac3b38a1 | QueryFinish | 2024-02-05 14:54:01.114439 | 2024-02-05 14:54:01.083110    |                 5 |                0 | b1c37b4c-3ef3-4026-8888-8facba026268
```

Logs on the initiator:
```
2024.02.05 14:54:01.083613 [ 2346057 ] {b1c37b4c-3ef3-4026-8888-8facba026268} <Debug> executeQuery: (from [::1]:37336)
...
2024.02.05 14:54:01.108058 [ 1700466 ] {b1c37b4c-3ef3-4026-8888-8facba026268} <Debug> Connection (host4:9000): Sent data for 2 scalars, total 2 rows in 6.429e-05 sec., 30901 rows/sec., 68.00 B (1.00 MiB/sec.), compressed 0.4594594594594595 times to148.00 B (2.18 MiB/sec.)
...
```

Logs on the shard
```
2024.02.05 14:54:01.108337 [ 3870854 ] {} <Trace> TCP-Session: cd600915-61d7-44d6-9d1c-db9be227c9c8 Creating query context from session context, user_id: 94309d50-4f52-5250-31bd-74fecac179db, parent context user: default
2024.02.05 14:54:01.108636 [ 3870854 ] {bec13910-2293-4226-bb2f-64cbac3b38a1} <Debug> executeQuery: (from [::ffff:10.246.128.9]:40804, initial_query_id: b1c37b4c-3ef3-4026-8888-8facba026268)
...
2024.02.05 14:54:01.114457 [ 3870854 ] {bec13910-2293-4226-bb2f-64cbac3b38a1} <Debug> executeQuery: Read 48407 rows, 553.28 KiB in 0.005984 sec., 8089405.080213904 rows/sec., 90.29 MiB/sec.
2024.02.05 14:54:01.114567 [ 3870854 ] {bec13910-2293-4226-bb2f-64cbac3b38a1} <Trace> HashJoin: Join data is being destroyed, 524288 bytes and 4375 rows in hash table
2024.02.05 14:54:01.114597 [ 3870854 ] {bec13910-2293-4226-bb2f-64cbac3b38a1} <Debug> MemoryTracker: Peak memory usage (for query): 725.65 KiB.
2024.02.05 14:54:01.114601 [ 3870854 ] {bec13910-2293-4226-bb2f-64cbac3b38a1} <Debug> TCPHandler: Processed in 0.00642317 sec.
```

It wasn't always like that. Here is an example from 22.8:
```
revision | type        | event_time_microseconds    | query_start_time_microseconds | query_id                             | initial_query_id                    
---------+-------------+----------------------------+-------------------------------+--------------------------------------+-------------------------------------
   54465 | QueryStart  | 2022-09-28 14:12:24.652869 | 2022-09-28 14:12:24.652869    | caa41952-ba5e-4b16-b02f-9c4a7a1cab15 | caa41952-ba5e-4b16-b02f-9c4a7a1cab15
   54465 | QueryFinish | 2022-09-28 14:12:25.153177 | 2022-09-28 14:12:24.652869    | caa41952-ba5e-4b16-b02f-9c4a7a1cab15 | caa41952-ba5e-4b16-b02f-9c4a7a1cab15
   54465 | QueryStart  | 2022-09-28 14:12:24.693056 | 2022-09-28 14:12:24.693056    | 5cf6d587-53e7-49f3-8317-5069048ff4e4 | caa41952-ba5e-4b16-b02f-9c4a7a1cab15
   54465 | QueryFinish | 2022-09-28 14:12:24.821581 | 2022-09-28 14:12:24.693056    | 5cf6d587-53e7-49f3-8317-5069048ff4e4 | caa41952-ba5e-4b16-b02f-9c4a7a1cab15
   54465 | QueryStart  | 2022-09-28 14:12:24.693617 | 2022-09-28 14:12:24.693617    | dcc4e39b-348c-443c-af23-4963f1e54b00 | caa41952-ba5e-4b16-b02f-9c4a7a1cab15
   54465 | QueryFinish | 2022-09-28 14:12:24.807704 | 2022-09-28 14:12:24.693617    | dcc4e39b-348c-443c-af23-4963f1e54b00 | caa41952-ba5e-4b16-b02f-9c4a7a1cab15
   54465 | QueryStart  | 2022-09-28 14:12:24.694179 | 2022-09-28 14:12:24.694179    | 1451941c-59a2-4915-b08f-fd36bf4e81fb | caa41952-ba5e-4b16-b02f-9c4a7a1cab15
   54465 | QueryFinish | 2022-09-28 14:12:25.151079 | 2022-09-28 14:12:24.694179    | 1451941c-59a2-4915-b08f-fd36bf4e81fb | caa41952-ba5e-4b16-b02f-9c4a7a1cab15
   54465 | QueryStart  | 2022-09-28 14:12:24.695337 | 2022-09-28 14:12:24.695337    | 34f1f6f4-3f24-4dc0-9549-97164c446176 | caa41952-ba5e-4b16-b02f-9c4a7a1cab15
   54465 | QueryFinish | 2022-09-28 14:12:24.810671 | 2022-09-28 14:12:24.695337    | 34f1f6f4-3f24-4dc0-9549-97164c446176 | caa41952-ba5e-4b16-b02f-9c4a7a1cab15
   54465 | QueryStart  | 2022-09-28 14:12:24.695458 | 2022-09-28 14:12:24.695458    | 3506703c-2445-4ba8-b13c-325f58280488 | caa41952-ba5e-4b16-b02f-9c4a7a1cab15
   54465 | QueryFinish | 2022-09-28 14:12:24.775568 | 2022-09-28 14:12:24.695458    | 3506703c-2445-4ba8-b13c-325f58280488 | caa41952-ba5e-4b16-b02f-9c4a7a1cab15
   54465 | QueryStart  | 2022-09-28 14:12:24.725217 | 2022-09-28 14:12:24.725217    | c0bf0038-c588-4dd5-aa1e-b2589918ce67 | caa41952-ba5e-4b16-b02f-9c4a7a1cab15
   54465 | QueryFinish | 2022-09-28 14:12:24.728257 | 2022-09-28 14:12:24.725217    | c0bf0038-c588-4dd5-aa1e-b2589918ce67 | caa41952-ba5e-4b16-b02f-9c4a7a1cab15
```

**Comment by alexey-milovidov** [comment]
@Algunenano, it's yours https://github.com/ClickHouse/ClickHouse/pull/51160

**Comment by SaltTan** [comment]
> @Algunenano, it's yours #51160

Actually this PR was merged in July, but I see the same unexpected behaviour in 23.4...

**Comment by SaltTan** [comment]
Unexpected in 23.1
```
┌─revision─┬─type────────┬────event_time_microseconds─┬─query_start_time_microseconds─┬─query_duration_ms─┬─is_initial_query─┬─initial_query_id─────────────────────┬─query_id─────────────────────────────┐
│    54470 │ QueryStart  │ 2024-02-06 14:43:26.799409 │    2024-02-06 14:43:26.799409 │                 0 │                1 │ 0785f967-ad46-4142-a5f4-be2306f876bc │ 0785f967-ad46-4142-a5f4-be2306f876bc │
│    54470 │ QueryStart  │ 2024-02-06 14:43:26.799409 │    2024-02-06 14:43:26.799409 │                 0 │                0 │ 0785f967-ad46-4142-a5f4-be2306f876bc │ 3f7eada7-2a10-4636-b850-6fc3f302717b │
│    54470 │ QueryFinish │ 2024-02-06 14:43:26.841455 │    2024-02-06 14:43:26.799409 │                29 │                0 │ 0785f967-ad46-4142-a5f4-be2306f876bc │ 3f7eada7-2a10-4636-b850-6fc3f302717b │
│    54470 │ QueryFinish │ 2024-02-06 14:43:26.857429 │    2024-02-06 14:43:26.799409 │                58 │                1 │ 0785f967-ad46-4142-a5f4-be2306f876bc │ 0785f967-ad46-4142-a5f4-be2306f876bc │

│    54470 │ QueryStart  │ 2024-02-06 14:43:21.483966 │    2024-02-06 14:43:21.483966 │                 0 │                1 │ db29dc78-b2a7-4178-808f-570ea1e5d1d1 │ db29dc78-b2a7-4178-808f-570ea1e5d1d1 │
│    54470 │ QueryStart  │ 2024-02-06 14:43:21.483966 │    2024-02-06 14:43:21.483966 │                 0 │                0 │ db29dc78-b2a7-4178-808f-570ea1e5d1d1 │ 49576fe4-8109-41e3-8611-bcef0e895a69 │
│    54470 │ QueryFinish │ 2024-02-06 14:43:21.505772 │    2024-02-06 14:43:21.483966 │                 4 │                0 │ db29dc78-b2a7-4178-808f-570ea1e5d1d1 │ 49576fe4-8109-41e3-8611-bcef0e895a69 │
│    54470 │ QueryFinish │ 2024-02-06 14:43:21.506500 │    2024-02-06 14:43:21.483966 │                22 │                1 │ db29dc78-b2a7-4178-808f-570ea1e5d1d1 │ db29dc78-b2a7-4178-808f-570ea1e5d1d1 │
└──────────┴─────────────┴────────────────────────────┴───────────────────────────────┴───────────────────┴──────────────────┴──────────────────────────────────────┴──────────────────────────────────────┘
```

**Comment by SaltTan** [comment]
> @amosbird, what do you think about this? Maybe #25022?

This one was merged in 2021, so included in 22.12

**Comment by Algunenano** [comment]
It was in fact due to https://github.com/ClickHouse/ClickHouse/pull/43455. The worst part is that apparently I did it consciously but I don't remember why, probably because I mistook query_start_time and initial_query_start_time. https://github.com/ClickHouse/ClickHouse/pull/59662 should fix it
